### PR TITLE
Correct torpedo store

### PR DESCRIPTION
**Pull request by dummyuser-study**
Corrected issues that the SonarCloud revealed.

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -30,18 +31,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (tempc4psgb) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:34]: This is correct a correct solution.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +31,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
-    double r = generator.nextDouble();
+    double r = rand.nextDouble();
```
Comment (tempc4psgb) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Nice solution here. We will have to incorporate this attribute in future changes

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +31,17 @@ public TorpedoStore(int numberOfTorpedos){
...
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (tempc4psgb) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:44]: This is correct

**Review by tempc4psgb — CHANGES_REQUESTED** [review]
Good fix.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -30,18 +31,17 @@ public TorpedoStore(int numberOfTorpedos){
...
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
```
Comment (tempc4psgb) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:39]: good change IMO

**Review by tempc4psgb — COMMENTED** [review]
good

**Comment by dummyuser-study** [comment]
Correct torpedo store

**Pull request merged**